### PR TITLE
Dashboard-112 [gototm] Updated accounts.freedom.tm

### PR DESCRIPTION
**Pull request by eshleebien**

### CODE REVIEW

File: app/Http/Controllers/Auth/AuthController.php [hunk 1]
```diff
@@ -38,40 +36,38 @@ public function getCheckpoint()
 
     public function getIndex()
     {
+        $state = str_random(5);
```
Comment (jekku) [inline app/Http/Controllers/Auth/AuthController.php]: Has little but considerable probability to create the same string when lots of users are in. What happens then?

Comment (eshleebien) [inline app/Http/Controllers/Auth/AuthController.php]: That's fine. it may generate same string but the sessions are different for each users

Comment (jekku) [inline app/Http/Controllers/Auth/AuthController.php]: Maybe around length 25+ the chances will become ridiculously smaller

File: app/Http/Controllers/Auth/AuthController.php [hunk 2]
```diff
@@ -38,40 +36,38 @@ public function getCheckpoint()
...
+        $tokens = request()->session()->get('auth.access_token');
+
+        try {
+            $response = request()->client->get(
+                'user', [
+                'headers' => [
+                    'Authorization'=> "Bearer $tokens->access_token"
+                ]]
+            );
+        } catch(ClientException $ex) {
```
Comment (jekku) [inline app/Http/Controllers/Auth/AuthController.php]: space after keywords

Comment (jekku) [inline app/Http/Controllers/Auth/AuthController.php]: Plus are you sure that $ex is always of type ClientException?

File: app/Http/Middleware/FreedomOauth.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        ];
+
+        $client = new Client(['base_uri' => config('oauth.base_uri')]);
+
+        try {
+            $response = $client->post(
+                'oauth/token',
+                ['form_params' => $data]
+            );
+        } catch(ClientException $ex) {
```
Comment (jekku) [inline app/Http/Middleware/FreedomOauth.php]: Same as above.

File: app/Http/Controllers/Auth/AuthController.php (hunk 1 above)
Comment (pandafox) [inline app/Http/Controllers/Auth/AuthController.php]: Why not just use a UUID if you need uniqueness and avoid the headache of guessing if it'll be unique or not?

Also, reviewboard is that way -> http://reviewboard.freedom.tm/

File: app/Http/Controllers/Auth/AuthController.php (hunk 2 above)
Comment (eshleebien) [inline app/Http/Controllers/Auth/AuthController.php]: yes. that's the only error expected here.

File: app/Http/Controllers/Auth/AuthController.php (hunk 1 above)
Comment (eshleebien) [inline app/Http/Controllers/Auth/AuthController.php]: no need. the goal of te state here is not for uniqueness (sessions are already unique). It's goal is to ensure the auth url is always fresh and used once (per session)

Comment (eshleebien) [inline app/Http/Controllers/Auth/AuthController.php]: @pandafox @jekku thoughts?

Comment (pandafox) [inline app/Http/Controllers/Auth/AuthController.php]: In that case I'd use time in millisecond instead of a random generated number.

Comment (jekku) [inline app/Http/Controllers/Auth/AuthController.php]: a linux epoch will do then :)

**Pull request merged**